react/components/atoms/lightbox/lightbox.js

```diff
@@ -69,10 +131,152 @@ export class Lightbox extends PureComponent {
+
+        this.translatedX = translateX;
+        this.translatedY = translateY;
+
+        this.setState({ zooming: true }, () => {
+            Animated.parallel([
+                Animated.timing(this.scaleRate, {
+                    toValue: this.fullZoomedInValue,
+                    duration: this.props.zoomAnimationDuration,
+                    useNativeDriver: true
+                }),
+                Animated.timing(this.state.translateX, {
+                    toValue: translateX,
+                    duration: this.props.translateAnimationDuration,
+                    useNativeDriver: true
+                }),
+                Animated.timing(this.state.translateY, {
+                    toValue: translateY,
+                    duration: this.props.translateAnimationDuration,
+                    useNativeDriver: true
+                })
+            ]).start(() => {
+                const scaleRatio = this.baseScale._value * 2;
+                this.lastScale = scaleRatio;
+                this.baseScale.setValue(this.lastScale);
+                this.scaleRate.setValue(1);
+                this.setState({ zooming: false, zoomed: true });
+            });
+        });
+    };
+
+    _doubleTapZoomOut = () => {
+        this.setState({ zooming: true }, () => {
+            Animated.parallel([
+                Animated.timing(this.scaleRate, {
+                    toValue: this.fullZoomedOutValue,
+                    duration: this.props.zoomAnimationDuration,
+                    useNativeDriver: true
+                }),
+                Animated.timing(this.baseScale, {
+                    toValue: 1,
+                    duration: this.props.zoomAnimationDuration,
+                    useNativeDriver: true
+                }),
+                Animated.timing(this.state.translateX, {
+                    toValue: 0,
+                    duration: this.props.zoomAnimationDuration,
+                    useNativeDriver: true
+                }),
+                Animated.timing(this.state.translateY, {
+                    toValue: 0,
+                    duration: this.props.zoomAnimationDuration,
+                    useNativeDriver: true
+                })
+            ]).start(() => {
+                this.lastScale = 1;
+                this.baseScale.setValue(this.lastScale);
+                this.setState({ zooming: false, zoomed: false });
+            });
+        });
+    };
+
     _imageSource = () => {
         return this.props.uri ? { uri: this.props.uri } : this.props.src;
     };
 
+    _getTranslateY = event => {
+        const y = event.y;
+        const middle = this.screenHeight / 2;
+        const touchMiddleDistance = middle - y;
+        const translateY =
+            Math.abs(touchMiddleDistance) > this.translatedYTreshold
+                ? this.translatedYTreshold
+                : touchMiddleDistance;
+
+        return y > middle ? -1 * Math.abs(translateY) : Math.abs(translateY);

```

Comment:
Perhaps simplify with the code below and also add comments about the logic behind this function
```suggestion
        const y = event.y;
        const middle = this.screenHeight / 2;
        const touchMiddleDistance = middle - y;
        const translateY = Math.min(this.translatedYTreshold, Math.abs(touchMiddleDistance))
        return y > middle ? -1 * translateY : translateY;
```